src/AutoMapper/TypeMap.cs

```diff
@@ -328,4 +338,39 @@ private void ApplyInheritedTypeMap(TypeMap inheritedTypeMap)
             }
         }
     }
+
+    public class TypeMap<TSource, TDestination>
+    {
+        private static readonly ConcurrentDictionary<IConfigurationProvider, TypeMap<TSource,TDestination>> Cache = new ConcurrentDictionary<IConfigurationProvider, TypeMap<TSource, TDestination>>();

```

Comment:
Because this type is generic, you get multiple caches per concrete generic type.